protostar/commands/format/formatting_summary.py

```diff
@@ -0,0 +1,135 @@
+from logging import Logger
+from typing import List
+from protostar.utils.log_color_provider import LogColorProvider
+from protostar.commands.format.formatting_result import (
+    FormattingResult,
+    BrokenFormattingResult,
+    CorrectFormattingResult,
+    IncorrectFormattingResult,
+)
+
+
+class FormatingSummary:
+    def __init__(
+        self, logger: Logger, check: bool = False, log_formatted: bool = False
+    ):
+        self._logger = logger
+        self._check_mode = check
+        self._log_formatted = log_formatted
+        self.broken: List[BrokenFormattingResult] = []
+        self.incorrect: List[IncorrectFormattingResult] = []
+        self.correct: List[CorrectFormattingResult] = []
```

Comment:
> Classes make functions visible while keeping data implied. Data structures make data visible while keeping functions implied.

https://blog.cleancoder.com/uncle-bob/2019/06/16/ObjectsAndDataStructures.html

Also, it would be nice to decouple Input/Output code from domain (let's not create another starknet_cli.py).

- `FormattingSummary` could be a `@dataclass`
- formatters could be functions that returns a string (use can use `log_color_provider` as a global variable)
  - you can create `format_formatting_result` 
- loggers should be used only in the `FormatCommand`